test/greenlight/test_test.clj

```diff
@@ -47,10 +49,35 @@
 (deftest optional-attr-map-test
   (let [system (component/system-map :greenlight.test-test/component 6)
         attr-test (blue/optional-attr-map)
-        test-result (test/run-test! system attr-test)]
+        test-result (test/run-test! system {} attr-test)]
     (is (= (::test/description test-result) "foobar"))
     (is (= (::test/context test-result) {:foo :bar}))
     (is (= :pass (::test/outcome test-result)))
     (is (= ["step-1"]
            (mapv ::step/title (::test/steps test-result))))
     (is (= 1 (count (::test/steps test-result))))))
+
+
+(defmacro with-io
+  [input & forms]
+  `(binding [*in* (io/reader (char-array ~input))
```

Comment:
You could also just make a `java.io.StringReader` directly here instead of the character-array copy from a string sequence.